### PR TITLE
Adds eating with utensils

### LINKED ISSUE #843: Allow eating with utensils.
74a4cc63cc6627fa3c7f21c28e60f20ba4e97ae3 added utensils. I only needed them as props for the map though so they don't do anything.

Being able to eat with them like SS13 would be neat.

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Closes #843 

Adds utensil component, with a configurable break chance (default: 0%, set to 20% for plastic utensils), break sound (default: Audio\items\snap.ogg), and kind (default: None).
Multiple kinds can be specified, in case anyone feels inclined to create a spork, forknife, or sporknife.
Food components can be restricted to a specific **combination** of utensils, but I have restricted none myself.
It checks for necessary utensils in the user's hands, but doesn't filter them or short-circuit as of the time of writing.
It won't check for utensils if none are needed, but if the food is directly clicked on with an active utensil it will still attempt to break it, even if it's not necessary.
Other entities can also be fed, and it takes into account the utensils that the user is holding.

Another change to food is that the trash that it leaves behind now only moves to the user's hands if they were already holding that food, otherwise it spawns where the food previously was.

### CODE REVIEW

File: Content.Client/Content.Client.csproj [hunk 1]
```diff
@@ -23,6 +23,9 @@
     <ProjectReference Include="..\RobustToolbox\Robust.Client\Robust.Client.csproj" />
     <ProjectReference Include="..\Content.Shared\Content.Shared.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="GameObjects\Components\Utensil" />
+  </ItemGroup>
```
Comment (Zumorica) [inline Content.Client/Content.Client.csproj:28]: Why did you add this? It's completely useless. The project automatically recognizes your cs files even in new folders.

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs [hunk 2]
```diff
@@ -63,7 +77,7 @@ void IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
             TryUseFood(eventArgs.User, eventArgs.Target);
         }
 
-        private bool TryUseFood(IEntity user, IEntity target)
+        internal bool TryUseFood(IEntity user, IEntity target, UtensilComponent utensilUsed = null)
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs:80]: At this point you could just make it public, since `Content.Server` isn't accessible from anywhere else.

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs [hunk 3]
```diff
@@ -102,19 +161,28 @@ private bool TryUseFood(IEntity user, IEntity target)
...
+                Owner.Delete();
+
+                // Put the trash in the user's hand
+                if (finisher.TryGetComponent(out ItemComponent item) &&
+                    handsComponent.CanPutInHand(item))
                 {
-                    handsComponent.PutInHand(itemComponent);
-                    return true;
+                    handsComponent.PutInHand(item);
+                    finisher.Transform.GridPosition = user.Transform.GridPosition;
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs:177]: Is this really necessary? Something tells me `PutInHand` should handle that.

File: Content.Client/Content.Client.csproj (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Client/Content.Client.csproj]: Good question, must have been by accident.

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs [hunk 4]
```diff
@@ -78,20 +92,65 @@ private bool TryUseFood(IEntity user, IEntity target)
...
+            var utensils = utensilUsed != null
+                ? new List<UtensilComponent> {utensilUsed}
+                : null;
+
+            if (_utensilsNeeded != UtensilKind.None)
+            {
+                utensils = new List<UtensilComponent>();
+                var kinds = UtensilKind.None;
+
+                if (user.TryGetComponent(out HandsComponent hands))
```
Comment (GlassEclipse) [inline Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs:109]: This check here would mean that if you were to have spaghetti and fork in your hand, you would just be able to click on the spaghetti instead of having to use the fork on the spaghetti.
I think it's more in the space station spirit to force users to click with the fork.

File: Content.Client/Content.Client.csproj [hunk 5]
```diff
@@ -23,6 +23,9 @@
     <ProjectReference Include="..\RobustToolbox\Robust.Client\Robust.Client.csproj" />
     <ProjectReference Include="..\Content.Shared\Content.Shared.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="GameObjects\Components\Utensil" />
```
Comment (GlassEclipse) [inline Content.Client/Content.Client.csproj]: This shouldn't be here.

File: Content.Shared/GameObjects/Components/Utensil/SharedUtensilComponent.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Robust.Shared.GameObjects;
+
+namespace Content.Shared.GameObjects.Components.Utensil
+{
+    [Flags]
+    public enum UtensilKind : byte
```
Comment (GlassEclipse) [inline Content.Shared/GameObjects/Components/Utensil/SharedUtensilComponent.cs:7]: I think UtensilType sounds way better. Kind sounds "kind"a weird

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs (hunk 4 above)
Comment (GlassEclipse) [reply Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs:109]: That's just my opinion, though

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs (hunk 3 above)
Comment (DrSmugleaf) [reply Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs]: It was a leftover from the user's hands being full and the trash being moved to the user's feet, which it doesn't do anymore. Changed it.

File: Content.Shared/GameObjects/Components/Utensil/SharedUtensilComponent.cs (hunk 6 above)
Comment (DrSmugleaf) [reply Content.Shared/GameObjects/Components/Utensil/SharedUtensilComponent.cs]: Changed it, originally named it after SuicideKind

File: Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs (hunk 4 above)
Comment (DrSmugleaf) [reply Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs:121]: Changed it, directly using food no longer works if it has utensil requirements. Also removed eating food when clicking on nothing with it, following how drinks work.
